doni/driver/worker/ironic.py

```diff
@@ -216,12 +216,15 @@ def process(
     ) -> "WorkerResult.Base":
         hw_props = hardware.properties
 
-        capabilities_string = ",".join(
-            [
-                f"{key}:{value}"
-                for key, value in hw_props.get("baremetal_capabilities").items()
-            ]
-        )
+        try:
+            capabilities_string = ",".join(
+                [
+                    f"{key}:{value}"
+                    for key, value in hw_props.get("baremetal_capabilities").items()
+                ]
+            )
+        except AttributeError:
```

Comment:
Added a check for being a Mapping, since otherwise the type might not have ".items()"

I'm always a bit fuzzy on the most "pythonic" way to handle this stuff

Reply:
I think if you know the shape of the thing, it's fine to just to truthy/falsy. Only when the inputs are really variable does it make sense to catch low-level errors, IMO. Like this:

```python
def try_int(num):
    try:
        return int(num)
    except ValueError:
        return None
```